src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -825,14 +836,24 @@ DagState uniqifyState(ImmutableArray<StateForCase> cases, ImmutableDictionary<Bo
                     // Select the next test to do at this state, and compute successor states
                     switch (state.SelectedTest = state.ComputeSelectedTest())
                     {
+                        case BoundDagAssignmentEvaluation e
+                            when state.RemainingValues.TryGetValue(e.Input, out IValueSet? currentValues) |
+                                 state.RemainingValues.TryGetValue(e.Target, out IValueSet? targetValues):
```

Comment:
nit: a line break here would help readability
Also, should `BoundDagAssignmentEvaluation` be added to `dumpDagTest`?